### PR TITLE
Enh clean config

### PR DESCRIPTION
**Pull request by tomMoral**
- [x] Refactor the client to have easier to maintain structure.
- [x] Add `clean` command to solve #125 
- [x] Add `config` command to make it easier to configurate `benchopt`. In particular, storing token, passing conda prefix if it is not in `PATH`, ...
- [x] Narrative doc for config.
- [x] Set permission for config file to user read/write only. Warn if it is not the case.

### CODE REVIEW

File: benchopt/config.py [hunk 1]
```diff
@@ -25,15 +23,36 @@
...
 
 
+def get_global_config_file():
+    "Return the global config file."
+
+    config_file = os.environ.get('BENCHO_CONFIG', None)
+    if config_file is not None:
+        config_file = Path(config_file)
+        assert config_file.exists(), (
+            f"BENCHO_CONFIG is set but file {config_file} does not exists."
```
Comment (agramfort) [inline benchopt/config.py:33]: I would rename to BENCHOPT_CONFIG

Comment (tomMoral) [reply benchopt/config.py]: ok I changed all environment variables to `BENCHOPT_*`.

**Review by agramfort — APPROVED** [review]

File: doc/config.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+.. _config_doc:
+
+BenchOpt configuration
+======================
+
+BenchOpt can be configured using setting files. These files can either be created directly or generated and modified using ``benchopt config``.
+
+There are two configuration levels. The first level is the global config for the ``benchopt`` client. It contains the system specific tweaks, the user info such as the *<GitHub token>* and the output levels. The second level is the configuration of the benchmarks. Each benchmark can have its own config for the kind of plots is displays by default and other display tweaks.
```
Comment (agramfort) [inline doc/config.rst:8]: It displays

File: doc/config.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+3. The default file is ``$HOME/.config/benchopt.ini``.
+
+For benchmark config files, they are usually located in the benchmark folder with name ``benchopt.ini``. If it does not exists, the default is to use the global config file.
+
+.. _config_structure:
+
+Config File Structure
+---------------------
+
+The config files for benchOpt follow the Microsoft Windows INI files structure. The global setting are grouped in a ``[benchopt]`` section:
```
Comment (agramfort) [inline doc/config.rst:34]: Check caps in benchopt word everywhere to be consistent

File: doc/doc-requirements.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 numpydoc
 sphinx-bootstrap-theme
-sphinx-click
+# sphinx-click
```
Comment (agramfort) [inline doc/doc-requirements.txt]: Clean?

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
looks good https://528-227720375-gh.circle-artifacts.com/0/dev/cli.html

**Pull request merged**